File: lang/ruby/src/fun/vac/algorithms/number_theory/Primality.rb
module Primality

    def self.isPrime(n)
        if n < 2
            return false
        end
        if (n & 1).zero? && n != 2
            return false
        end

        3.step(Math.sqrt(n).to_i(), 2) do |i|
            if (n % i).zero?
                return false
            end
        end

        true
    end

    def self.isComposite(n)
        n > 1 && !isPrime(n)
    end
end

File: go/muse/algorithms/number_theory/primality/primality.go
package primality

import "math"

func IsPrime(n int64) bool {
	if n < 2 {
		return false
	}
	if (n&1) == 0 && n != 2 {
		return false
	}

	for i, bound := int64(3), int64(math.Sqrt(float64(n)))+1; i < bound; i += 2 {
		if n%i == 0 {
			return false
		}
	}

	return true
}

func IsComposite(n int64) bool {
	return n > 1 && !IsPrime(n)
}
